packages/stark-ui/src/modules/pretty-print/components/pretty-print.component.spec.ts

```diff
@@ -0,0 +1,706 @@
+		'{"value": "Close", "onclick": "CloseDoc()"}]}}'
+	].join("");
+
+	const formattedJsonData: string = [
+		"{",
+		'  "menu": {',
+		'    "id": "file",',
+		'    "value": "File",',
+		'    "menuitem": [',
+		"      {",
+		'        "value": "New",',
+		'        "onclick": "CreateNewDoc()"',
+		"      },",
+		"      {",
+		'        "value": "Open",',
+		'        "onclick": "OpenDoc()"',
+		"      },",
+		"      {",
+		'        "value": "Close",',
+		'        "onclick": "CloseDoc()"',
+		"      }",
+		"    ]",
+		"  }",
+		"}"
+	].join("\n"); // should contain line breaks
+
+	const rawJavaScriptData: string = [
+		"function calculateData(seed, operationFn) {" +
+			"var data = operationFn(seed);" +
+			"if (!data){" +
+			"data = 'could not calculate data';" +
+			"}" +
+			"return data;" +
+			"}"
+	].join("");
+
+	const rawTypeScriptData: string = [
+		"function calculateData(seed:any, operationFn:Function):any {" +
+			"var data:any = operationFn(seed);" +
+			"if (!data){" +
+			"data = 'could not calculate data';" +
+			"}" +
+			"return data;" +
+			"}"
+	].join("");
+
+	/**
+	 * async beforeEach
+	 */
+	beforeEach(async(() => {
+		return TestBed.configureTestingModule({
+			declarations: [StarkPrettyPrintComponent, TestHostComponent],
+			providers: [
+				{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() },
+				{ provide: STARK_ROUTING_SERVICE, useClass: MockStarkRoutingService }
+			],
+			schemas: [NO_ERRORS_SCHEMA] // tells the Angular compiler to ignore unrecognized elements and attributes
+		}).compileComponents();
+	}));
+
+	/**
+	 * Synchronous beforeEach
+	 */
+	beforeEach(() => {
+		hostFixture = TestBed.createComponent(TestHostComponent);
+		hostComponent = hostFixture.componentInstance;
+		hostFixture.detectChanges(); // trigger initial data binding
+
+		component = hostComponent.prettyPrintComponent;
+	});
+
+	describe("on initialization", () => {
+		it("should set internal component properties", () => {
+			expect(hostFixture).toBeDefined();
+			expect(component).toBeDefined();
+
+			expect(component.logger).not.toBeNull();
+			expect(component.logger).toBeDefined();
+			expect(component.routingService).not.toBeNull();
+			expect(component.routingService).toBeDefined();
+		});
+
+		it("should NOT have any inputs set", () => {
+			expect(component.data).toBeUndefined();
+			expect(component.format).toBeUndefined();
+			expect(component.enableHighlighting).toBeUndefined();
+		});
+	});
+
+	describe("Formatting", () => {
+		describe("XML", () => {
+			beforeEach(() => {
+				hostComponent.data = rawXmlData;
+				hostComponent.format = "xml";
+				hostFixture.detectChanges();
+			});
+
+			it(shouldHaveInputs, () => {
+				expect(component.data).toBe(rawXmlData);
+				expect(component.format).toBe("xml");
+				expect(component.enableHighlighting).toBeUndefined();
+			});
+
+			it("should nicely format raw XML data", () => {
+				let formattedData: string = component.prettyString;
```

Comment:
Apart from asserting the formatted data from the `prettyString` variable, I think you should also assert the actual content of the HTML `<pre>` element in the same way we did in the old Stark. In this way we test the actual rendering of the component. What do you think?

The same for all the other tests in this file ;)